README.md

```diff
@@ -1,46 +1,64 @@
 
-This project builds on top of [@devcontainers/cli](https://www.npmjs.com/package/@devcontainers/cli)
+A similar [Azure DevOps Task](./docs/azure-devops-task.md) is also available!
 
+Note that this project builds on top of [@devcontainers/cli](https://www.npmjs.com/package/@devcontainers/cli) which can be used in other automation systems.
 
 ## GitHub Action
+The examples below show usage of the GitHub Action - see the [GitHub Action documentation](./docs/github-action.md) for more details.
 
-The example below shows usage of the GitHub Action - see the [GitHub Action documentation](./docs/github-action.md) for more details:
+> **NOTE:** This Action is not currently capable of taking advantage of pre-built Codespaces. However, pre-built images are supported.
 
 
+**Pre-building an image:**
+
 ```yaml
-- name: Build and run dev container task
+- name: Pre-build dev container image
   uses: devcontainers/ci@v0.2
   with:
     imageName: ghcr.io/example/example-devcontainer
-    runCmd: make ci-build
+    cacheFrom: ghcr.io/example/example-devcontainer
+    push: true
 ```
 
-## Azure DevOps Task
-
-The example below shows usage of the Azure DevOps Task - see the [Azure DevOps Task documentation](./docs/azure-devops-task.md) for more details:
+**Using a dev container for a CI build:**
 
 ```yaml
-- task: DevcontainersCI@0
-  inputs:
-    imageName: 'yourregistry.azurecr.io/example-dev-container'
-    runCmd: 'make ci-build'
-    sourceBranchFilterForPush: refs/heads/main
+- name: Run make ci-build in dev container
+  uses: devcontainers/ci@v0.2
+  with:    
+    # [Optional] Even if you're not directly referencing a pre-built image,
+    # you can still use the image as a cache for your Dockerfile build.
+    cacheFrom: ghcr.io/example/example-devcontainer
+
+    push: never
+    runCmd: make ci-build
 ```
 
+**Both at once:**
+
+```yaml
+- name: Pre-build image and run make ci-build in dev container
+  uses: devcontainers/ci@v0.2
+  with:
+    imageName: ghcr.io/example/example-devcontainer
+    cacheFrom: ghcr.io/example/example-devcontainer
+    push: true
+    runCmd: make ci-build
+```
 
 ## CHANGELOG
 
 ### Version 0.2.0
 
 This version updates the implementation to use [@devcontainers/cli](https://www.npmjs.com/package/@devcontainers/cli).
 
-This brings many benefits around compatibility with VS Code Dev Containers. One key area is that [container-features](https://code.visualstudio.com/docs/remote/containers#_dev-container-features-preview) can now be used in CI.
+This brings many benefits around compatibility with Dev Containers. One key area is that dev container [Features](https://code.visualstudio.com/docs/remote/containers#_dev-container-features-preview) can now be used in CI.
```

Comment:
Does it still make sense to link to VS Code docs here, or is something like containers.dev/features a better target?